datafusion/physical-expr/src/equivalence/properties.rs

```diff
@@ -1524,6 +1524,102 @@ impl Hash for ExprWrapper {
     }
 }
 
+/// Take sort orderings for unioned sides of equal length, and return the unioned ordering.
+///
+/// Example:
+///     child1 = order by a0, b, c
+///     child2 = order by a, b, c
+///     => union's joint order is a0, a, b, c.
+fn calculate_joint_ordering(
+    lhs: &EquivalenceProperties,
```

Comment:
I didn't actually fully understand why do you need such a function. I think its logic is not correct.
SCHEMA: a-b-c-a0
CHILD1: (5,15,25,35),(4,15,25,36),(3,15,25,37)
CHILD2: (5,15,25,35),(6,15,25,34),(7,15,25,33)
How could we deduce unioned output is (a0, a, b, c)?